src/two_finger_grasp_data.cpp

```diff
@@ -0,0 +1,214 @@
+
+  // Search within the sub-namespace of this end effector name
+  ros::NodeHandle child_nh(nh, end_effector);
+
+  error += !rosparam_shortcuts::get(parent_name, child_nh, "gripper_finger_width", gripper_finger_width_);
+  error += !rosparam_shortcuts::get(parent_name, child_nh, "max_grasp_width", max_grasp_width_);
+  error += !rosparam_shortcuts::get(parent_name, child_nh, "max_finger_width", max_finger_width_);
+  error += !rosparam_shortcuts::get(parent_name, child_nh, "min_finger_width", min_finger_width_);
+  rosparam_shortcuts::shutdownIfError(parent_name, error);
+
+  return true;
+}
+
+bool TwoFingerGraspData::setGraspWidth(double fraction_open, double min_finger_width,
+                                       trajectory_msgs::JointTrajectory& grasp_posture)
+{
+  if (fraction_open < 0 || fraction_open > 1)
+  {
+    ROS_ERROR_STREAM_NAMED("grasp_data", "Invalid fraction passed in " << fraction_open);
+    return false;
+  }
+
+  // Ensure min_finger_width is not less than actual min finger width
+  double min_finger_width_adjusted = std::max(min_finger_width, min_finger_width_);
+
+  // Max width = max_finger_width_
+  // Min width = min_finger_width_adjusted
+  double distance_btw_fingers =
+      min_finger_width_adjusted + (max_finger_width_ - min_finger_width_adjusted) * fraction_open;
+  return fingerWidthToGraspPosture(distance_btw_fingers, grasp_posture);
+}
+
+bool TwoFingerGraspData::fingerWidthToGraspPosture(double distance_btw_fingers,
+                                                   trajectory_msgs::JointTrajectory& grasp_posture)
+{
+  // TODO(mlautman): Change this function to take in a method for translating joint values to grasp width
+  //       Currently this function simply interpolates between max open and max closed
+  ROS_DEBUG_STREAM_NAMED("grasp_data", "Setting grasp posture to have distance_between_fingers of "
+                                           << distance_btw_fingers);
+
+  // Error check
+  if (distance_btw_fingers > max_finger_width_ + std::numeric_limits<double>::epsilon() ||
+      distance_btw_fingers < min_finger_width_ - std::numeric_limits<double>::epsilon())
+  {
+    ROS_DEBUG_STREAM_NAMED("grasp_data", "Requested " << distance_btw_fingers << " is beyond limits of "
+                                                      << min_finger_width_ << "," << max_finger_width_);
+    return false;
+  }
+
+  // NOTE: We assume a linear relationship between the actuated joint values and the distance between fingers.
+  //       This is probably incorrect but until we expose an interface for passing in a function to translate from
+  //       joint values to grasp width, it's the best we got...
+  // TODO(mlautman): Make it so that a user can pass in a function here.
+  std::vector<std::string> joint_names = pre_grasp_posture_.joint_names;
+  std::vector<double> grasp_pose = grasp_posture_.points[0].positions;
+  std::vector<double> pre_grasp_pose = pre_grasp_posture_.points[0].positions;
```

Comment:
If these are just shorter names for convenience maybe make them const references